test/test-ipld.js

```diff
@@ -0,0 +1,33 @@
+/* globals describe, it, before */
+import assert from 'assert'
+import { encode, decode, mkGetBlock, writer } from '../src/ipld.js'
+
+describe('ipld', () => {
+  let block
+  before(async () => {
+    block = await encode({ anything: 'goes' })
+  })
+  it('encode', () => {
+    assert.equal(block.cid.toString(), 'bafyreif6wvo5txz35pv5vnungtazalvmmdhes3hopaqopjb63gvwwovlwy')
+    assert.equal(block.value.anything, 'goes')
+    assert.equal(block.bytes.length, 15)
+  })
+
+  it('decode', async () => {
+    // console.log(block)
+    // const byteView = new DataView(block.bytes.buffer, 0)
+    const decoded = decode(block.bytes, block.cid)
```

Comment:
it does not like `block.bytes` which is a Uint8Array, the TS declaration for the argument type is:

```
export type ByteView<T> =
  | Uint8Array
  | Uint8Array & { data: T }
```

Not sure how to convert

Reply:
is it missing `Buffer`?

i don’t use TS so you may have to ask irakli

Reply:
I tried `block.bytes.buffer` which is the underlying array buffer, but that didn't work also.